compileFile.js

```diff
@@ -2,34 +2,40 @@ const { spawn } = require("child_process");
 const preferences = require("./preferencesHelper");
 const path = require("path");
 const locationHelper = require("./locationHelper");
+const config = require("./config")
+
+/**
+ * Get flags which needed for compile based on language
+ * @param {string} language programming language
+ * @param {string} filePath complete path to .c, .cpp or .py file
+ */
+function getFlags(language, filePath) {
+    const ext = config.extensions[language];
+    let flags = preferences().get("compilationFlags" + ext).split(" ");
+    if (flags[0] === "")
+        flags = [];
+
+    if (language === 'Python')
+        flags = ['-m', 'compileall', '-b', filePath, ...flags];
+    else {
+        // type of compiler ( g++ or gcc )
+        const outputLocation = locationHelper.getBinLocation(language, filePath);
+        flags = [filePath, "-o", outputLocation, ...flags];
+    }
+    return flags
+}
 
 // compile the file and return a promise with the result/error
```

Comment:
Let's make this a JSDoc, ie a comment with ```/* ... */```.

Reply:
Do you mean comment about what each flag does?

Reply:
Yep.